README.md

```diff
@@ -89,4 +89,20 @@ version topoSemverVersion("prefix")
 ```
 where prefix is the prefix for the tags you want to search (see Prefix tags).
 
-Topological sorting will then find the closest tag to HEAD that also has matches the prefix, and use that tag as the base for the version. 
\ No newline at end of file
+Topological sorting will then find the closest tag to HEAD that also has matches the prefix, and use that tag as the base for the version.
+
+## Version Object
+
+The plugin's version methods return a ``SemverVersion`` object and not a ``String``.  It can be used as a parameter for Gradle's `version`.  Making this an object allows for pulling out pieces of the version string for other uses (if desired).
+
+```java
+class SemverVersion {
+  String toString()  // The full version string
+  String getTagVersion() // The matched tagged
+  String getHeadCommitHash() // The git commit hash of the HEAD commit
+  Integer getCommitCount() // The number of commits HEAD is from the matched tag
+  Integer getBuildNumber() // The value of the BUILD_NUMBER environment variable
+  boolean isDirty() // Is the git repo dirty?
+  boolean isVersionStableRelease() // Does the full version represent a stable version?
```

Comment:
The only reason I put it is so they know if it's a "clean version" since all the data will get filled.  Kind of like flexVersion.  If you're a release, it's just TAGNAME but you still have commit count, hash, etc.

We can remove it though if you feel we should.